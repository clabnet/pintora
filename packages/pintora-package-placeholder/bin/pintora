#!/usr/bin/env node

const str = `
This is a placeholder package, in case the name 'pintora' is taken by other projects.

Pintora is an extensible text-to-diagrams library that works in both browser and node.js.

Please check out https://pintorajs.vercel.app/docs/getting-started/usage/ for more instructions of using @pintora/cli in Node.js.
`

console.log(str)
