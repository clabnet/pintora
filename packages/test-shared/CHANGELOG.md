# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.1.0-alpha.18](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.17...v0.1.0-alpha.18) (2022-01-16)


### Features

* **pintora-cli:** read cli args `--theme` ([a502c35](https://github.com/hikerpig/pintora/commit/a502c35cf1dbbad65c5847e2752e17711b8dccef))





# [0.1.0-alpha.11](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.10...v0.1.0-alpha.11) (2021-12-04)


### Features

* **pintora-diagrams:** Add basic activityDiagram [#3](https://github.com/hikerpig/pintora/issues/3) ([718a80e](https://github.com/hikerpig/pintora/commit/718a80e55d209d618fb855a463a151b880bf6fc3))





# [0.1.0-alpha.8](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.7...v0.1.0-alpha.8) (2021-08-29)


### Features

* [sequenceDiagram] add custom group color ([f93fe8a](https://github.com/hikerpig/pintora/commit/f93fe8a6fdb81af55eeb6f303f426c6150eeaf86))





# [0.1.0-alpha.4](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.3...v0.1.0-alpha.4) (2021-08-15)


### Bug Fixes

* **pintora-diagrams:** [componentDiagram] avoid interface node overlapping ([213cea0](https://github.com/hikerpig/pintora/commit/213cea090f40717390417e143366026d841f9808))


### Features

* **pintora-diagrams:** [erDiagram] add support for key coloumn like 'PK' ([64dad3d](https://github.com/hikerpig/pintora/commit/64dad3ddd8cefb912d7c83e14cf8a2dc0e64c5f8))





# [0.1.0-alpha.0](https://github.com/hikerpig/pintora/compare/v0.0.1...v0.1.0-alpha.0) (2021-08-01)


### Features

* Add componentDiagram ([c967f1c](https://github.com/hikerpig/pintora/commit/c967f1c9a969a813adac9a9967a06a024f04d73f))
