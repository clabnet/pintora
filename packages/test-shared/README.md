# `@pintora/test-shared`

> TODO: description

## Usage

```
const testShared = require('@pintora/test-shared');

// TODO: DEMONSTRATE API
```
