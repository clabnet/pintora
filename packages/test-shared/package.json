{
  "name": "@pintora/test-shared",
  "version": "0.0.0",
  "description": "Shared testing utils for pintora",
  "keywords": [
    "pintora"
  ],
  "author": "hikerpig <hikerpigwinnie@gmail.com>",
  "homepage": "https://github.com/hikerpig/pintora#readme",
  "license": "MIT",
  "main": "lib/index.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/hikerpig/pintora.git"
  },
  "scripts": {
    "compile": "tsc",
    "watch": "tsc -w"
  },
  "bugs": {
    "url": "https://github.com/hikerpig/pintora/issues"
  }
}
