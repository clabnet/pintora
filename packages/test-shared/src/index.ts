export { EXAMPLES } from './data/examples'
export { stripStartEmptyLines } from './util'
