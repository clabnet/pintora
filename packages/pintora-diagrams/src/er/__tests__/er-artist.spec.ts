import pintora from '@pintora/core'
import { EXAMPLES } from '@pintora/test-shared'
import { testDraw, prepareDiagramConfig } from '../../__tests__/test-util'
import { erDiagram } from '../index'

describe('component-artist', () => {
  beforeAll(() => {
    prepareDiagramConfig()
    pintora.diagramRegistry.registerDiagram('erDiagram', erDiagram)
  })

  it('will not throw error', () => {
    expect(testDraw(EXAMPLES.erLarge.code).graphicIR).toBeTruthy()
  })
})
