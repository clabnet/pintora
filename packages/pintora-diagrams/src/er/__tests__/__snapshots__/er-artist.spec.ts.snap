// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`er-artist will draw inheritance 1`] = `
Object {
  "mark": Object {
    "attrs": Object {},
    "children": Array [
      Object {
        "children": Array [],
        "class": "er__relations",
        "type": "group",
      },
      Object {
        "attrs": Object {
          "id": "person",
        },
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 2,
              "stroke": "#3b4044",
              "text": "",
            },
            "class": "er__entity-box",
            "itemId": "entity-person",
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "fontWeight": "bold",
              "id": "entity-person",
              "text": "person",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "er__entity-label",
            "itemId": "entity-person",
            "type": "text",
          },
          Object {
            "attrs": Object {},
            "children": Array [
              Object {
                "attrs": Object {},
                "children": Array [
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-person-attr-1-type",
                      "text": "int",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-person-attr-1-name",
                      "text": "age",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                ],
                "type": "group",
              },
              Object {
                "attrs": Object {},
                "children": Array [
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-person-attr-2-type",
                      "text": "string",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-person-attr-2-name",
                      "text": "phone_number",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                ],
                "type": "group",
              },
            ],
            "itemId": "entity-person",
            "type": "group",
          },
        ],
        "class": "er__entity",
        "type": "group",
      },
      Object {
        "attrs": Object {
          "id": "customer",
        },
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 2,
              "stroke": "#3b4044",
              "text": "",
            },
            "class": "er__entity-box",
            "itemId": "entity-customer",
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "fontWeight": "bold",
              "id": "entity-customer",
              "text": "customer",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "er__entity-label",
            "itemId": "entity-customer",
            "type": "text",
          },
          Object {
            "attrs": Object {},
            "children": Array [
              Object {
                "attrs": Object {},
                "children": Array [
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-customer-attr-1-type",
                      "text": "string",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-customer-attr-1-name",
                      "text": "address",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-customer-attr-1-name",
                      "text": "deliver address",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                ],
                "type": "group",
              },
              Object {
                "attrs": Object {},
                "children": Array [
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-customer-attr-2-key",
                      "text": "PK",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-customer-attr-2-type",
                      "text": "string",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 11.9,
                      "id": "entity-customer-attr-2-name",
                      "text": "id",
                      "textAlign": "left",
                      "textBaseline": "middle",
                    },
                    "class": "er__entity-label",
                    "type": "text",
                  },
                  Object {
                    "attrs": Object {
                      "fill": "#fffbf9",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                ],
                "type": "group",
              },
            ],
            "itemId": "entity-customer",
            "type": "group",
          },
        ],
        "class": "er__entity",
        "type": "group",
      },
      Object {
        "attrs": Object {
          "id": "deliverer",
        },
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 2,
              "stroke": "#3b4044",
              "text": "",
            },
            "class": "er__entity-box",
            "itemId": "entity-deliverer",
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "fontWeight": "bold",
              "id": "entity-deliverer",
              "text": "deliverer",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "er__entity-label",
            "itemId": "entity-deliverer",
            "type": "text",
          },
          Object {
            "attrs": Object {},
            "children": Array [],
            "itemId": "entity-deliverer",
            "type": "group",
          },
        ],
        "class": "er__entity",
        "type": "group",
      },
      Object {
        "attrs": Object {},
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fffbf9",
              "lineJoin": "round",
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          Object {
            "attrs": Object {
              "fill": "#fffbf9",
              "lineJoin": "round",
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "ISA",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "er__relationship-label",
            "type": "text",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "ISA",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "er__relationship-label",
            "type": "text",
          },
          Object {
            "attrs": Object {
              "lineJoin": "round",
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          Object {
            "attrs": Object {
              "lineJoin": "round",
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          Object {
            "attrs": Object {
              "lineJoin": "round",
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          Object {
            "attrs": Object {
              "lineJoin": "round",
              "stroke": "#3b4044",
            },
            "type": "path",
          },
        ],
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;
